UniversityApp.BussinessLogic/Services/StudentsService.cs

```diff
@@ -30,6 +32,43 @@ public async Task<List<StudentDomainModel>> GetStudents()
             return result;
         }
 
+        public async Task<(List<StudentDomainModel> Students, int TotalRecordCount)> SearchStudents(StudentSearchModel student, int pageNo, int pageSize)
+        {
+            int skip = (pageNo - 1) * pageSize;
+
+            Expression<Func<Student, bool>> ContainsFirstname = s => s.FirstName.Contains(student.FirstName);
+            Expression<Func<Student, bool>> ContainsLastName = s => s.LastName.Contains(student.LastName);
+            Expression<Func<Student, bool>> IsAge = s => s.Age == student.Age;
+            Expression<Func<Student, bool>> IsGender = s => s.Gender.Contains(student.Gender);
+
+            var predicates = new List<Expression<Func<Student, bool>>>();
+
+            if(!String.IsNullOrEmpty(student.FirstName))
+            {
+                predicates.Add(ContainsFirstname);
```

Comment:
I think it would be more readable without defining predicate before defining if statement:
                predicates.Add( s => s.FirstName.Contains(student.FirstName));

As this logic doesn't contain any business logic it could go torepository layer now.